core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationManagerImpl.java

```diff
@@ -107,24 +109,32 @@ void start() {
          }
       }
 
+      // Install the listener
+      GlobalConfigurationStateListener stateCacheListener = new GlobalConfigurationStateListener(this);
+      getStateCache().addListener(stateCacheListener, new ScopeFilter(CACHE_SCOPE));
+
       // Initialize caches which are present in the initial state. We do this before installing the listener.
+      log.debugf("Starting caches from global state: %s", getStateCache().keySet());
       for (Map.Entry<ScopedState, Object> e : getStateCache().entrySet()) {
          if (CACHE_SCOPE.equals(e.getKey().getScope())) {
             String cacheName = e.getKey().getName();
             CacheState cacheState = (CacheState) e.getValue();
             uncheckedAwait(createCacheLocally(cacheName, cacheState));
          }
       }
-      // Install the listener
-      GlobalConfigurationStateListener stateCacheListener = new GlobalConfigurationStateListener(this);
-      getStateCache().addListener(stateCacheListener, new ScopeFilter(CACHE_SCOPE));
 
-      // Tell the LocalConfigurationManager that it can load any persistent caches
+      // Load any persistent caches that are not in the state cache
+      log.debugf("Starting caches from local persisted state: %s", persistedConfigurations.keySet());
       List<CompletableFuture<Configuration>> all = persistedConfigurations.entrySet().stream().map((entry) ->
             // The cache configuration was permanent, it still needs to be
             getOrCreateCache(entry.getKey(), entry.getValue(), EnumSet.noneOf(CacheContainerAdmin.AdminFlag.class))
       ).collect(Collectors.toList());
       uncheckedAwait(CompletableFutures.sequence(all));
+
+      Set<String> staticCacheNames = new HashSet<>(configurationManager.getDefinedCaches());
```

Comment:
Why are you not starting the static caches before the persisted ones ?

Reply:
It was a shortcut to avoid starting a cache with an incompatible configuration, since we already check compatibility for caches from local persistent state. But I'm now convinced that I need to replicate the compatibility check for statically defined caches anyway.